src/labone/server/server.py

```diff
@@ -131,4 +134,48 @@ async def start_local_mock(
     reader = await capnp.AsyncIoStream.create_connection(sock=read)
     writer = await capnp.AsyncIoStream.create_connection(sock=write)
     # create server for the local socket pair
-    return capnp_server_factory(writer, schema, mock), reader
+    return capnp_server_factory(writer, schema, server), reader
+
+
+def start_server(
+    schema: CapnpStructReader,
+    server: CapnpServer,
+) -> None:
+    """Start the server.
+
+    Start the capnp server with the given schema and server instance.
+    The server address and port can be specified via command line arguments.
+    The server keeps running until it is interrupted.
+
+    Args:
+        schema: Parsed capnp schema (`reflection_capnp.CapSchema`).
+        server: The concrete server implementation.
+    """
+
+    def _host_port() -> tuple[str, int]:
+        parser = argparse.ArgumentParser(
+            usage="""Runs the server bound to the given address/port ADDRESS. """,
+        )
+
+        parser.add_argument("address", help="ADDRESS:PORT")
+
+        return parser.parse_args().address.split(":")
+
+    async def _new_connection(
+        stream: capnp.AsyncIoStream,
+    ) -> None:
+        await capnp_server_factory(
+            stream=stream,
+            schema=schema,
+            server=server,
+        ).on_disconnect()
+
+    async def _run_server() -> None:
+        await ensure_capnp_event_loop()
+        host, port = _host_port()
+        server = await capnp.AsyncIoStream.create_server(_new_connection, host, port)
+        async with server:
+            await server.serve_forever()
+
+    with suppress(KeyboardInterrupt):
+        asyncio.run(capnp.run(_run_server()))
```

Comment:
question: Is the `capnp.run` really necessary? In the `_run_server` function we call `ensure_capnp_event_loop` right at the beginning which should take care of it already right?

Reply:
Good point. It's indeed not needed.